dask/dataframe/tests/test_dataframe.py

```diff
@@ -2630,31 +2631,85 @@ def test_to_dask_array_unknown(as_frame):
     assert all(np.isnan(x) for x in result)
 
 
+@pytest.mark.skipif(
+    not PANDAS_GT_110, reason=f"No extension arrays in pandas version {pd.__version__}"
+)
+@pytest.mark.parametrize("lengths", [None, [2, 3], True])
+@pytest.mark.parametrize("as_frame", [False, True])
 @pytest.mark.parametrize(
-    "lengths,as_frame,meta",
+    "input_func, dtype, na_value, meta",
     [
-        ([2, 3], False, None),
-        (True, False, None),
-        (True, False, np.array([], dtype="f4")),
+        (
+            lambda: [1, 2, 3, 4, 5],
+            None,
+            no_default,
+            None,
+        ),
+        (
+            lambda: [1, 2, 3, 4, 5],
+            "float32",
+            no_default,
+            None,
+        ),
+        (
+            lambda: [1, 2, 3, 4, 5],
+            float,
+            no_default,
+            np.array([], dtype="f4"),
+        ),
+        (
+            lambda: pd.arrays.SparseArray([np.nan, 1, 2, np.nan, 1]),
+            None,
+            no_default,
+            None,
+        ),
+        (
+            lambda: pd.arrays.SparseArray([np.nan, 1, 2, np.nan, 1]),
+            int,
+            0,
+            None,
+        ),
+        (
+            lambda: pd.arrays.SparseArray([np.nan, 1, 2, np.nan, 1]),
+            None,
+            0.0,
+            np.array([], dtype="float64"),
+        ),
     ],
 )
-def test_to_dask_array(meta, as_frame, lengths):
-    s = pd.Series([1, 2, 3, 4, 5], name="foo", dtype="i4")
-    a = dd.from_pandas(s, chunksize=2)
+def test_to_dask_array(input_func, dtype, na_value, meta, as_frame, lengths):
+    from dask.array.utils import assert_eq
```

Comment:
Since `dask.dataframe` depends on NumPy we can move this `import` to the top of the module. Or, since we already have `import dask.array as da`, we could use `da.utils.assert_eq` here